src/index.js

```diff
@@ -138,10 +138,15 @@ class Dropzone extends React.Component {
     this.isFileDialogActive = false;
 
     fileList.forEach((file) => {
-      if (this.fileAccepted(file) && this.fileMatchSize(file)) {
-        if (!disablePreview) {
+      if (!disablePreview) {
+        try {
           file.preview = window.URL.createObjectURL(file); // eslint-disable-line no-param-reassign
+        } catch (err) {
+          // Continue regardless of error
```

Comment:
So, using invariant actually breaks the test I added: 'it should not throw an error when failing to build a preview'.

I'm not sure this is the right use-case for invariant.